repose-aggregator/components/filters/uri-stripper-filter/src/main/resources/META-INF/schema/config/uri-stripper.xsd

```diff
@@ -65,14 +71,160 @@
                 </xs:documentation>
             </xs:annotation>
         </xs:attribute>
+    </xs:complexType>
+
+    <xs:complexType name="LinkResource">
+        <xs:annotation>
+            <xs:documentation>
+                <html:p>
+                    A specific ReST resource that contains one or more links which should have the stripped URI
+                    path segment re-inserted.
+                </html:p>
+            </xs:documentation>
+        </xs:annotation>
+
+        <xs:all>
+            <xs:element name="request" type="HttpMessage" minOccurs="0" maxOccurs="1"/>
+            <xs:element name="response" type="HttpMessage" minOccurs="0" maxOccurs="1"/>
+        </xs:all>
```

Comment:
@Mario-Lopez It really depends on the configuration. In your first example, the `response` element is redundant. In your second example, the `link-response` element _would_ be redundant if you also defined transformations on the request for the same resources.

The funny thing is that I asked @adrianjgeorge about this exact thing this morning.

Having `request`/`response` be children of `link-resource` is consistent with what exists in the body-patcher filter where `request`/`response` are children of `change`.

Personally, I would have to see the common use case to be able to determine which is more appropriate.